qa/tasks/cephfs/test_volumes.py

```diff
@@ -2075,8 +1915,51 @@ def test_subvolume_clone_in_progress_getpath(self):
         subvolpath = self._get_subvolume_path(self.volname, clone)
         self.assertNotEqual(subvolpath, None)
 
-        # now, unprotect snapshot
-        self._fs_cmd("subvolume", "snapshot", "unprotect", self.volname, subvolume, snapshot)
+        # remove snapshot
+        self._fs_cmd("subvolume", "snapshot", "rm", self.volname, subvolume, snapshot)
+
+        # verify clone
+        self._verify_clone(subvolume, clone)
+
+        # remove subvolumes
+        self._fs_cmd("subvolume", "rm", self.volname, subvolume)
+        self._fs_cmd("subvolume", "rm", self.volname, clone)
+
+        # verify trash dir is clean
+        self._wait_for_trash_empty()
+
+    def test_subvolume_clone_in_progress_snapshot_rm(self):
+        subvolume = self._generate_random_subvolume_name()
+        snapshot = self._generate_random_snapshot_name()
+        clone = self._generate_random_clone_name()
+
+        # create subvolume
+        self._fs_cmd("subvolume", "create", self.volname, subvolume)
+
+        # do some IO
+        self._do_subvolume_io(subvolume, number_of_files=64)
+
+        # snapshot subvolume
+        self._fs_cmd("subvolume", "snapshot", "create", self.volname, subvolume, snapshot)
+
+        # schedule a clone
+        self._fs_cmd("subvolume", "snapshot", "clone", self.volname, subvolume, snapshot, clone)
+
+        # snapshot should not be deletable now
+        try:
+            self._fs_cmd("subvolume", "snapshot", "rm", self.volname, subvolume, snapshot)
+        except CommandFailedError as ce:
+            if ce.exitstatus != errno.EAGAIN:
+                raise RuntimeError("invalid error code when removing source snapshot of a clone")
```

Comment:
I think self.fail is preferred now,
https://github.com/ceph/ceph/pull/35417#discussion_r441902597

Reply:
Ack! Will make the change (in both places).

Just checking: There are ~88 more instances of the same in this file, should we push a cleanup to change those as well in the near future? So that the pattern is not applied again in a future commit?

Reply:
I don't think we need to do in this PR. Of course you may.

Reply:
> I think self.fail is preferred now,
> [#35417 (comment)](https://github.com/ceph/ceph/pull/35417#discussion_r441902597)

I think self.assert is preferred first and then self.fail ?

self.assertEqual(ce.exitstatus, errno.EAGAIN)

Reply:
lgtm otherwise

Reply:
Addressed. (both fail and assertEqual parts)